js/classes/entities/foreground/ui/button.js

```diff
@@ -32,7 +31,8 @@ class Button {
                     this.removeFromWorld = true;
                 }
             }
-        });
+        }
+        document.body.addEventListener("click", this.listener);
```

Comment:
Before, I was assigning `this.listener` the return value of `addEventListener`, which makes no sense because `addEventListener` returns `undefined`. This was a problem because `this.listener` was being passed to `removeEventListener`, which didn't do anything because `removeEventListener` expects the listener function as its second argument. That's why `this.listener` is assigned the listener function itself now.